packages/babel-plugin-proposal-decorators/src/transformer-2023-05.ts

```diff
@@ -1211,6 +1260,223 @@ function createLocalsAssignment(
   return t.assignmentExpression("=", lhs, rhs);
 }
 
+function isProtoSetter(
+  node: t.Identifier | t.StringLiteral | t.BigIntLiteral | t.NumericLiteral,
+) {
+  return node.type === "Identifier"
+    ? node.name === "__proto__"
+    : node.value === "__proto__";
+}
+
+function isDecorated(node: t.Class | ClassDecoratableElement) {
+  return node.decorators && node.decorators.length > 0;
+}
+
+function shouldTransformElement(node: ClassElement) {
+  switch (node.type) {
+    case "ClassAccessorProperty":
+      return true;
+    case "ClassMethod":
+    case "ClassProperty":
+    case "ClassPrivateMethod":
+    case "ClassPrivateProperty":
+      return isDecorated(node);
+    default:
+      return false;
+  }
+}
+
+function shouldTransformClass(node: t.Class) {
+  return (
+    isDecorated(node) ||
+    node.body.body.some(node => shouldTransformElement(node))
```

Comment:
very nit, but we can skip an extra function
```suggestion
    node.body.body.some(shouldTransformElement)
```